dev/integration_tests/android_semantics_testing/lib/src/tests/controls_page.dart

```diff
@@ -28,15 +28,15 @@ class _SelectionControlsPageState extends State<SelectionControlsPage> {
   bool _isLabeledOn = false;
   int _radio = 0;
 
-  void _updateCheckbox(bool newValue) {
+  void _updateCheckbox(bool? newValue) {
     setState(() {
-      _isChecked = newValue;
+      _isChecked = newValue!;
     });
   }
 
-  void _updateRadio(int newValue) {
+  void _updateRadio(int? newValue) {
```

Comment:
It seems like this should not be nullable, since is always asserted to be non-null. Either that, or below it shouldn't assert, but rather set to zero or something if the value is null.

Reply:
The function signature is ValueChange(T?), It can be null if the radioButton is toggle-able. I can add an assert, but I think it is redundant since i already force casted it below

Reply:
No, I don't mean add an assert (I was calling the `!` an assertion that it is non-null, which it is, but it's not an "assert", sorry for being confusing).

I'm just saying that if the argument is nullable, and you always force the value to be non-null, then perhaps either the argument should be also non-nullable (which I understand is different than the signature for `Radio.onChanged`), or the value shouldn't be always forced to non-nullable, and be something like `_radio = newValue ?? 0` instead of `_radio = newValue!`.

Reply:
It can be nullable for the different use cases of RadioButton, but It can't be null in the RadioButton use case in this file.

If it is null here, that means something is wrong in the code in this file. I think a force cast would be more preferable in this case?

Reply:
Yeah,OK, I guess since it's a test it's probably fine.